tests/Serializer/JsonSerializationTest.php

```diff
@@ -144,16 +142,16 @@ public function getFirstClassListCollectionsValues()
     public function testFirstClassListCollections($items, $expected): void
     {
         $collection = new FirstClassListCollection($items);
-
-        self::assertSame($expected, $this->serialize($collection));
+        $serialized = $this->serialize($collection);
+        self::assertSame($expected, $serialized);
+        self::assertEquals($collection, $this->deserialize($serialized, get_class($collection)));
     }
 
     public function getFirstClassMapCollectionsValues()
     {
         return [
             [[1, 2, 3], '{"0":1,"1":2,"2":3}'],
             [[], '{}'],
-            [["a" => "b", "c" => "d", 5], '{"a":0,"c":0,"0":5}'],
```

Comment:
why this have been removed?

Reply:
I was a bit confused about this examples that cannot be deserialised into initial data. I'll bring it back and create new data set.

Reply:
I've splitted implementation into separate methods. I hope it make sense now.